app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -394,104 +464,130 @@ class PipelineSampleReport extends React.Component {
-      const key_parts = filter_key.split('_');
-      const val = (taxon[key_parts[0]] || {})[key_parts[1]];
-      if (val < threshold) {
-        // console.log([val, threshold, filter_key])
-        return false;
-      }
-    }
-    return true;
-  }
-
-  applyThresholdFilters(candidate_taxons, play_animation = true) {
-    let thresholded_taxons = [];
-    let genus_taxon = {};
-    let matched_taxons = [];
-    for (let i = 0; i < candidate_taxons.length; i++) {
-      const taxon = candidate_taxons[i];
-      if (taxon.genus_taxid == taxon.tax_id) {
-        // genus
-        if (matched_taxons.length > 0) {
-          thresholded_taxons.push(genus_taxon);
-          thresholded_taxons = thresholded_taxons.concat(matched_taxons);
-        } else if (this.taxonPassThresholdFilter(genus_taxon)) {
-          thresholded_taxons.push(genus_taxon);
-        }
-        genus_taxon = taxon;
-        matched_taxons = [];
-      } else {
-        // species
-        if (this.taxonPassThresholdFilter(taxon)) {
-          matched_taxons.push(taxon);
-        }
-      }
-    }
+  appendThresholdFilter() {
+    const stateCopy = Object.assign([], this.state.activeThresholds);
+    stateCopy.push({ label: '', operator: '', value: '' });
+    this.setState({
+      activeThresholds: stateCopy
+    });
+  }
 
-    if (matched_taxons.length > 0) {
-      thresholded_taxons.push(genus_taxon);
-      thresholded_taxons = thresholded_taxons.concat(matched_taxons);
-    } else if (this.taxonPassThresholdFilter(genus_taxon)) {
-      thresholded_taxons.push(genus_taxon);
+  removeThresholdFilter(pos) {
+    const stateCopy = Object.assign([], this.state.activeThresholds);
+    stateCopy.splice(pos, 1);
+    this.setState({
+      activeThresholds: stateCopy
+    });
+  }
+
+  isThresholdValid(threshold) {
+    if (threshold.hasOwnProperty('label')
+      && threshold.hasOwnProperty('operator')
+      && threshold.hasOwnProperty('value')) {
+        return ((threshold.label.length > 0)
+        && (threshold.operator.length > 0) 
+        && (threshold.value != '' && !isNaN(threshold.value)));
     }
+    return false;
+  }
+
+  saveThresholdFilters() {
+    // prevent saving threshold with invalid values
+    const activeThresholds = this.state.activeThresholds.filter((threshold) => {
+      return this.isThresholdValid(threshold);
+    });
+    window.localStorage.setItem('activeThresholds', JSON.stringify(activeThresholds));
+    $('.advanced-filters-modal').slideUp(300);
+    this.applyThresholdFilters(this.state.taxonomy_details, this.state.activeThresholds, true);
+  }
 
-    this.applySearchFilter(0, this.state.excluded_categories, thresholded_taxons);
+  getSavedThresholdFilters() {
+    const activeThresholds = window.localStorage.getItem('activeThresholds');
+    return (activeThresholds) ? JSON.parse(activeThresholds) : [];
+  }
 
-    if (play_animation) {
-      this.flash();
+  applyThresholdFilters(candidateTaxons, rules, animate = true) {
```

Comment:
I think there are way too many changes than just move the filters to the top.

Reply:
Yeah  because we also had to remove the sidebar and re-style form inputs ui and implement the advanced filtering